Model/Giftcard/Response/Giftcard.php

```diff
@@ -136,26 +139,117 @@ public function set(TransactionResponse $response, CartInterface $quote)
     }
 
     /**
-     * Get error message
+     * Save group transaction data
      *
-     * @return mixed|string|null
+     * @return void
      */
-    public function getErrorMessage()
+    protected function saveGroupTransaction()
     {
-        if ($this->response->isSuccess()) {
-            return null;
+        $this->groupTransaction->saveGroupTransaction($this->response->data());
+    }
+
+    /**
+     * Create order from quote
+     *
+     * @return AbstractExtensibleModel|OrderInterface|object|null
+     * @throws LocalizedException
+     */
+    protected function createOrderFromQuote($success = true)
```

Comment:
what's the use of the $success variable, it doesn't describe its purpose

Reply:
I used that variable to know when I should create an order. We should create order only at the first success transaction, if the first transaction is failed we don't want to create it.

I will remove that variable because I can get the value using $this->response->isSuccess()